Installation issue: hip@4.0.0

<!-- Thanks for taking the time to report this build failure. To proceed with the report please:

1. Title the issue "Installation issue: <name-of-the-package>".
2. Provide the information required below.

We encourage you to try, as much as possible, to reduce your problem to the minimal example that still reproduces the issue. That would help us a lot in fixing it quickly and effectively! -->

### Steps to reproduce the issue

<!-- Fill in the exact spec you are trying to build and the relevant part of the error message -->
```console
$ spack install hip@4.0.0
==> Installing hip-4.0.0-5o7bjupmptlwhreq5vdxghtpr6g2u3nw
==> No binary for hip-4.0.0-5o7bjupmptlwhreq5vdxghtpr6g2u3nw found: installing from source
==> Fetching https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/_source-cache/archive/d7/d7b78d96cec67c55b74ea3811ce861b16d300410bc687d0629e82392e8d7c857.tar.gz
######################################################################## 100.0%
==> Applied patch /home/myusername/spack/var/spack/repos/builtin/packages/hip/0001-Make-it-possible-to-specify-the-package-folder-of-ro.patch
==> Applied patch /home/myusername/spack/var/spack/repos/builtin/packages/hip/0003-Improve-compilation-without-git-repo.3.10.0.patch
==> Applied patch /home/myusername/spack/var/spack/repos/builtin/packages/hip/0004-Drop-clang-rt-builtins-linking-on-hip-host.3.10.0.patch
==> Ran patch() for hip
==> hip: Executing phase: 'cmake'
==> Error: ProcessError: Command exited with status 1:
    'cmake' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:STRING=/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hip-4.0.0-5o7bjupmptlwhreq5vdxghtpr6g2u3nw' '-DCMAKE_BUILD_TYPE:STRING=RelWithDebInfo' '-DCMAKE_INTERPROCEDURAL_OPTIMIZATION:BOOL=OFF' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=OFF' '-DCMAKE_INSTALL_RPATH:STRING=/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hip-4.0.0-5o7bjupmptlwhreq5vdxghtpr6g2u3nw/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hip-4.0.0-5o7bjupmptlwhreq5vdxghtpr6g2u3nw/lib64;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/comgr-4.0.0-kcgpexabbnrrklkxkpfnxug4ov5cb4wj/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/ncurses-6.2-76vyzwtyv2utfq7vttyruuant5u7fuek/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/z3-4.8.9-jwzytxuxjmmnnclhxkrx7jmdzcug6cqu/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/zlib-1.2.11-ix3si5dzsvolel7tprmmiy2qnn435wni/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hsa-rocr-dev-4.0.0-ocmo4q2n7o6i33c72yrm7xtsoyejgo2i/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/numactl-2.0.14-gwl45jjvy4vi5ozehgez7tawcwmjxc43/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libelf-0.8.13-5cefspksznzky5ajo4ff5wt5irimsvva/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/llvm-amdgpu-4.0.0-evkr7qyibfwkmnbfv5c5ohz53nfznjvs/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hwloc-2.4.1-6j357x4dkp643b5kv7xlkqec6yefuv5h/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libpciaccess-0.16-r27ovdbgiskizcwwt463znaipiv2itcw/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libxml2-2.9.10-5aaujrhwses77xglrsswkrknt4yakgym/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libiconv-1.16-4s4y3kiopmt4uhoblns3if2hcirvyhlp/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/xz-5.2.5-v7lveanhbynm5vjwrat6jq7ye3xv65wt/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/mesa18-18.3.6-wfvva4x5mxwvgdoebokpsftdff2irl6s/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/expat-2.2.10-pc6x4pfvtqoam3662dt5edyfx3u3s23f/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libbsd-0.10.0-a7mbqw76bg7qaicwfnpiwhhytoppalui/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/glproto-1.4.17-egod7anupisnwumojyjupdbvounvkww6/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libx11-1.7.0-prm7ecpkpuon6dmpl5uoy7cirqbg2hv2/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/inputproto-2.3.2-rs3fs5oo4p22tldxhdzw66jcortcoqmf/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/kbproto-1.0.7-5ejq56b4i5vyfsgnvbvbutyv57q6znty/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libxcb-1.14-i2dwodd2foegxdfqh2d75dle3dvwxrgl/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libpthread-stubs-0.4-si3fr5etnklqvrxobf4zs5ssj3ff6zue/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libxau-1.0.8-papkbkwoxydymapua5azeo6x3ktvh2ra/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/xproto-7.0.31-plhrjrb6cept2xax7gkzekz73iap75im/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libxdmcp-1.1.2-ln7ubv2iv5al4ymionhmwcjauvtqu7wq/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/xcb-proto-1.14.1-ukpgx5iaoyhnnvcjgtlyx4imkqjfoolx/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/xextproto-7.3.0-5ljniv4f4sadtzl7hyodzsi3f4h6g4ft/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/libxext-1.3.3-wl23yawvmfgsnpemkk5qfajmh7mepllk/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/rocm-device-libs-4.0.0-h6z5yg3xwmprzskjthfxmosq63ldanzd/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/texinfo-6.5-exejbs23jhmgdl4dmbrcqohozuyd2jfw/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/perl-5.32.1-slfmbbfmie5uphm22m64tejs777ye7hc/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/berkeley-db-18.1.40-ifa4cj7zbtsnj3aw6hppovf7skf7tetn/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/gdbm-1.18.1-neyjkf5fj3kebempbncneuegkata4aae/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/readline-8.0-jd7txpuxwaz4wmeu2yjwlz54ejo7ejrw/lib;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/comgr-4.0.0-kcgpexabbnrrklkxkpfnxug4ov5cb4wj/lib64;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hsakmt-roct-4.0.0-zrvaopyxbi5j5lwvzvefg6lyqrx7cjlr/lib64' '-DCMAKE_PREFIX_PATH:STRING=/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/cmake-3.19.5-mk6hvvv6denvmdjj4ito3gdfxmigd3g5;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/perl-5.32.1-slfmbbfmie5uphm22m64tejs777ye7hc;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/mesa18-18.3.6-wfvva4x5mxwvgdoebokpsftdff2irl6s;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hip-rocclr-4.0.0-yipkcntmopxhtvvzuc6gbzhjg5um7bug;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hsakmt-roct-4.0.0-zrvaopyxbi5j5lwvzvefg6lyqrx7cjlr;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hsa-rocr-dev-4.0.0-ocmo4q2n7o6i33c72yrm7xtsoyejgo2i;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/comgr-4.0.0-kcgpexabbnrrklkxkpfnxug4ov5cb4wj;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/llvm-amdgpu-4.0.0-evkr7qyibfwkmnbfv5c5ohz53nfznjvs;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/rocm-device-libs-4.0.0-h6z5yg3xwmprzskjthfxmosq63ldanzd;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/rocminfo-4.0.0-3hhaiwfbnan5orvoipxr3j72gjvxg2nk;/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/numactl-2.0.14-gwl45jjvy4vi5ozehgez7tawcwmjxc43' '-DHIP_COMPILER:STRING=clang' '-DHIP_PLATFORM:STRING=rocclr' '-DHSA_PATH:STRING=/home/myusername/spack/opt/spack/linux-scientific7-skylake_avx512/gcc-8.3.1/hsa-rocr-dev-4.0.0-ocmo4q2n7o6i33c72yrm7xtsoyejgo2i' '-DHIP_RUNTIME:STRING=ROCclr' '/tmp/myusername/spack-stage/spack-stage-hip-4.0.0-5o7bjupmptlwhreq5vdxghtpr6g2u3nw/spack-src'

1 error found in build log:
     9     -- Detecting C compile features - done
     10    -- Detecting CXX compiler ABI info
     11    -- Detecting CXX compiler ABI info - done
     12    -- Check for working CXX compiler: /home/myusername/spack/lib/spack/env/gcc/g++ - skipped
     13    -- Detecting CXX compile features
     14    -- Detecting CXX compile features - done
  >> 15    CMake Error at CMakeLists.txt:41 (string):
     16      string sub-command REPLACE requires at least four arguments.
     17    
     18    
     19    Using CPACK_DEBIAN_PACKAGE_RELEASE local
     20    CPACK_RPM_PACKAGE_RELEASE: local%{?dist}
     21    -- HIP Platform: rocclr

See build log for details:
  /tmp/myusername/spack-stage/spack-stage-hip-4.0.0-5o7bjupmptlwhreq5vdxghtpr6g2u3nw/spack-build-out.txt

...
```

### Information on your system

<!-- Please include the output of `spack debug report` -->
* **Spack:** 0.16.1-1661-fc15f85
* **Python:** 3.6.8
* **Platform:** linux-scientific7-skylake_avx512
* **Concretizer:** original
<!-- If you have any relevant configuration detail (custom `packages.yaml` or `modules.yaml`, etc.) you can add that here as well. -->

### Additional information

<!-- Please upload the following files. They should be present in the stage directory of the failing build. Also upload any config.log or similar file if one exists. -->
* [spack-build-out.txt](https://github.com/fmani/host_files/blob/main/spack-build-out.txt?raw=true)
* [spack-build-env.txt](https://github.com/fmani/host_files/blob/main/spack-build-env.txt?raw=true)

<!-- Some packages have maintainers who have volunteered to debug build failures. Run `spack maintainers <name-of-the-package>` and @mention them here if they exist. -->
@arjun-raj-kuppala @srekolam 

### General information


<!-- These boxes can be checked by replacing [ ] with [x] or by clicking them after submitting the issue. -->
- [x] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [x] I have run `spack maintainers <name-of-the-package>` and @mentioned any maintainers
- [x] I have uploaded the build log and environment files
- [x] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/hip/package.py
+++ b/var/spack/repos/builtin/packages/hip/package.py
@@ -4,6 +4,7 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 from spack.util.prefix import Prefix
+from spack.hooks.sbang import filter_shebang
 import os
 
 
@@ -189,6 +190,10 @@ class Hip(CMakePackage):
                 'hipconfig', 'hipify-cmakefile'
             ]
             filter_file(match, substitute, *files, **kwargs)
+
+            # This guy is used during the cmake phase, so we have to fix the
+            # shebang already here in case it is too long.
+            filter_shebang('hipconfig')
 
         if '@3.7.0:' in self.spec:
             numactl = self.spec['numactl'].prefix.lib

